### ISSUE TITLE
clj-kondo not detecting deprecated .cljc namespaces in report and cache

### ISSUE THREAD

**Issue by escherize** [clj-kondo/clj-kondo#2568]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

repro'd on `clj-kondo v2025.06.05` and `clj-kondo v2025.04.07`

**macro usage**
n/a

**platform**

Running on the JVM (`clojure -M:kondo --lint`) using https://github.com/metabase/metabase/blob/master/deps.edn#L459

running from brew installation, too

**editor**
n/a

**problem**

clj-kondo does not properly cache the deprecated status for .cljc namespaces when marked as deprecated via namespace metadata.

**repro**

https://github.com/escherize/kondo_cljc_deprecation_repro

tldr is:

Setup files:
- `src/x.clj` - deprecated via ns metadata
- `src/y.cljc` - deprecated via ns metadata  
- `src/z.clj` - requires both x and y namespaces

Run: `clj-kondo --cache true --lint src/x.clj src/z.clj src/y.cljc`

Output shows only warning for x namespace:
```
src/z.clj:3:5: warning: Namespace x is deprecated.
linting took 11ms, errors: 0, warnings: 1
```

Examining the cache files:

`cat .clj-kondo/.cache/v1/clj/x.transit.json` shows:
```json
{
  "x": {
    "row": 3,
    "col": 1,
    "fixed-arities": [0],
    "name": "x",
    "ns": "x",
    "top-ns": "x",
    "type": "fn"
  },
  "deprecated": true,
  "filename": "src/x.clj"
}
```

`cat .clj-kondo/.cache/v1/cljc/y.transit.json` shows:
```json
{
  "filename": "src/y.cljc",
  "clj": {
    "y": {
      "row": 3,
      "col": 1,
      "fixed-arities": [0],
      "name": "y",
      "ns": "y",
      "top-ns": "y",
      "type": "fn"
    }
  },
  "cljs": {
    "y": {
      "row": 3,
      "col": 1,
      "fixed-arities": [0],
      "name": "y",
      "ns": "y",
      "top-ns": "y",
      "type": "fn"
    }
  }
}
```

**config**
No custom configuration used.

**expected behavior**

The y.cljc namespace should be marked as deprecated in the cache (missing `"deprecated": true` field), and clj-kondo should warn about using the deprecated y namespace in addition to the x namespace warning.
